infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/LogicalGroupLogEntryReader.java

```diff
@@ -12,10 +34,97 @@
+
+        if (txEntryStreamIds.contains(CLIENT_CONFIG_TABLE_ID) &&
+                isCurrentSessionImpacted(entry.getEntries().get(CLIENT_CONFIG_TABLE_ID))) {
+            log.info("Group destination change detected, log entry sync will be stopped and a new snapshot sync " +
+                    "will be triggered！");
+            throw new GroupDestinationChangeException();
+        }
+
+        return super.isValidTransactionEntry(entry);
+    }
+
+    /**
+     * Helper method for checking if the smr entries present in the opaque stream (that current session's log entry
+     * reader is tracking) contain any group destination change with respect to current session. The opaque entries
+     * will be deserialized to verify if the Sink cluster of current session is among the target destinations.
+     *
+     * @param groupConfigTableEntries SMREntries of LogReplicationModelMetadataTable.
+     * @return True if current session is impacted by group destination config change, false otherwise.
+     */
+    private boolean isCurrentSessionImpacted(List<SMREntry> groupConfigTableEntries) {
+        Set<String> groups = ((LogReplicationLogicalGroupConfig) replicationContext.getConfig(session))
+                .getLogicalGroupToStreams().keySet();
+
+        for (SMREntry smrEntry : groupConfigTableEntries) {
+            // Get serialized form of arguments for registry table. They were sent in OpaqueEntry and
+            // need to be deserialized using ProtobufSerializer
+            Object[] objs = smrEntry.getSMRArguments();
+            ByteBuf keyBuf = Unpooled.wrappedBuffer((byte[]) objs[0]);
+            ClientDestinationInfoKey clientInfoKey = (ClientDestinationInfoKey) protobufSerializer
+                    .deserialize(keyBuf, null);
+
+            if (clientInfoKey.getModel() != session.getSubscriber().getModel() ||
+                    !clientInfoKey.getClientName().equals(session.getSubscriber().getClientName())) {
+                // SMREntry for other clients
+                continue;
+            }
+
+            ByteBuf valueBuf = Unpooled.wrappedBuffer((byte[]) objs[1]);
+            CorfuRecord<DestinationInfoVal, Message> destinations =
+                    (CorfuRecord<DestinationInfoVal, Message>) protobufSerializer.deserialize(valueBuf, null);
+
+            // From a session's point of view, there are 2 ways it could be impacted:
+            // (1) a new group is added to have current session's Sink as its destination
+            // (2) an existing group (already in config) no longer has current session's Sink as its destination
+            boolean isGroupAdded = !groups.contains(clientInfoKey.getGroupName()) &&
+                    destinations.getPayload().getDestinationIdsList().contains(session.getSinkClusterId());
+            boolean isGroupRemoved = groups.contains(clientInfoKey.getGroupName()) &&
+                    !destinations.getPayload().getDestinationIdsList().contains(session.getSinkClusterId());
```

Comment:
we should also consider the case of a group being deleted (key is deleted).  If not in this PR, let us add it in a follow-up PR.

Reply:
Yes you are right, in the client implementation we do `deleteRecord` in when removing destinations and the destination list is empty after removal. I will add a check here

Reply:
is this check added on L106?
`else if (CorfuStreamEntry.getOperationType(smrEntry).equals(OperationType.DELETE) &&
                    groups.contains(clientInfoKey.getGroupName())) {
                    log.info("Group {} has no destinations now and is removed from client metadata table.",
                            clientInfoKey.getGroupName());
                    return true;`

Reply:
yes, my question was for the DELETE op case.  Went through the code in more detail and it getting handled in L106.  Thanks.